pretixscan/app/src/main/java/eu/pretix/pretixscan/droid/PretixScan.kt

```diff
@@ -31,6 +34,23 @@ class PretixScan : MultiDexApplication() {
     var flipperInit: FlipperInitializer.IntializationResult? = null
     lateinit var connectivityHelper: ConnectivityHelper
 
+    private fun migrateSqlCipher() {
+        val dbPass = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) KeystoreHelper.secureValue(KEYSTORE_PASSWORD, true)
+        else KEYSTORE_PASSWORD
+
+        System.loadLibrary("sqlcipher")
+
+        val databaseFile = getDatabasePath(Models.DEFAULT.name)
+        SQLiteDatabase.openOrCreateDatabase(databaseFile, dbPass, null, null, object: SQLiteDatabaseHook {
+            override fun preKey(connection: SQLiteConnection) {
+            }
+
+            override fun postKey(connection: SQLiteConnection) {
+                connection.execute("PRAGMA cipher_migrate;", emptyArray(), null)
```

Comment:
```suggestion
                val result = connection.executeForLong("PRAGMA cipher_migrate;", emptyArray(), null)
                if (result != 0L) {
                    throw SQLiteException("cipher_migrate failed")
                }
```

> android.database.sqlite.SQLiteException: unknown error (code 0): Queries can be performed using SQLiteDatabase query or rawQuery methods only.

See https://github.com/sqlcipher/sqlcipher-android/issues/6 and https://github.com/sqlcipher/sqlcipher-android/blob/master/sqlcipher/src/androidTest/java/net/zetetic/database/sqlcipher_cts/MigrateDatabaseFrom1xFormatToCurrentFormatTest.java#L28 - we should check the return value here